src/Arrays/MPIStateArrays.jl

```diff
@@ -529,30 +526,23 @@ using Requires
 # better names, for example `device` is also defined in CUDAdrv
 
 device(::Union{Array, SArray, MArray}) = CPU()
+device(::CuArray) = CUDA()
 device(Q::MPIStateArray) = device(Q.data)
 
 realview(Q::Union{Array, SArray, MArray}) = Q
+realview(Q::CuArray) = Q
 realview(Q::MPIStateArray) = Q.realdata
 
-@init @require CuArrays = "3a865a2d-5b23-5a0f-bc46-62713ec82fae" begin
-  using .CuArrays
-  using .CuArrays.CUDAnative
-
-  device(::CuArray) = CUDA()
-  realview(Q::CuArray) = Q
-
-  # transform all arguments of `bc` from MPIStateArrays to CuArrays
-  # and replace CPU function with GPU variants
-  function transform_broadcasted(bc::Broadcasted, ::CuArray)
-    transform_cuarray(bc)
-  end
-
-  function transform_cuarray(bc::Broadcasted)
-    Broadcasted(CuArrays.cufunc(bc.f), transform_cuarray.(bc.args), bc.axes)
-  end
-  transform_cuarray(mpisa::MPIStateArray) = mpisa.realdata
-  transform_cuarray(x) = x
+# transform all arguments of `bc` from MPIStateArrays to CuArrays
+# and replace CPU function with GPU variants
+function transform_broadcasted(bc::Broadcasted, ::CuArray)
+  transform_cuarray(bc)
+end
+function transform_cuarray(bc::Broadcasted)
+  Broadcasted(CuArrays.cufunc(bc.f), transform_cuarray.(bc.args), bc.axes)
```

Comment:
I am not sure what this code intends to do, but this should probably be `GPUifyLoops.contextualize` and might also want to use `Adapt.jl`?

Reply:
Me either, I just un-indented it. Outside the scope of this PR, but feel free to address.